### PR TITLE
Apply env var collection path prefixes in shell integration scripts when mac+login shell

### LINKED ISSUE #99878: Prepending PATH env var with environmentVariableCollection doesn't work on macOS
This was covered a little bit in https://github.com/microsoft/vscode/issues/94153 but that issue is locked and can't be discussed there. This feature feels pretty broken on macOS and I hope might be revisited.

The issue is on macOS prepending to `PATH`:

```
context.environmentVariableCollection.prepend("PATH", "/foo/my/path:");
```

This ends up being added to the end of the path, rather than the start. @Tyriar wrote up why this is in microsoft/vscode-docs@38653d4 and offered two workarounds, however neither of them are usable in my testing (as well as not really being feasible to tell all users to change):

- `"terminal.integrated.inheritEnv": false` makes no difference at all, the PATH till is added to the end (this is a problem because the purpose of using `prepend` above is that we need to override any existing SDK on PATH with the one the user just selected)
- `"terminal.integrated.shellArgs": []` results in a completely empty `PATH` besides the values I've prepended, which means that pretty much everything else is missing and won't work. This might be solvable by pushing `process.env.PATH` into `environmentVariableCollection` too, however the prompt to the user would look ridiculous as it would include their entire PATH

I appreciate this probably isn't VS Code's fault, but given macOS is fairly common and this behaviour seems to happen on a clean macOS install (it happens on both of mine, and I've customised almost nothing) it feels like it's worth trying to fix (maybe even if it means sending `export PATH=foo:$PATH` once the shell is initialised? 😄).

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #99878

pwsh doesn't need the fix which is why it's excluded.

### CODE REVIEW

File: src/vs/platform/terminal/common/environmentVariable.ts [hunk 1]
```diff
@@ -12,8 +14,47 @@ export interface IEnvironmentVariableMutator {
...
+
+	/**
+	 * Applies this collection to a process environment.
+	 * @param variableResolver An optional function to use to resolve variables within the
+	 * environment values.
+	 */
+	applyToProcessEnvironment(env: IProcessEnvironment, variableResolver?: VariableResolver): Promise<void>;
+
+	/**
+	 * Generates a diff of this connection against another. Returns undefined if the collections are
```
Comment (meganrogge) [inline src/vs/platform/terminal/common/environmentVariable.ts:56]:
```suggestion
	 * Generates a diff of this collection against another. Returns undefined if the collections are
```

File: src/vs/platform/terminal/node/terminalEnvironment.ts [hunk 2]
```diff
@@ -230,6 +236,39 @@ export function getShellIntegrationInjection(
 	return undefined;
 }
 
+/**
+ * On macOS the profile calls path_helper which adds a bunch of standard bin directories to the
+ * beginning of the PATH. This causes significant problems for the environment variable
+ * collection API as the custom paths added to the end will now be somewhere in the middle of
+ * the PATH. To combat this, VSCODE_PATH_PREFIX is used to re-apply any prefix after the profile
+ * has run. This will cause duplication in the PATH but should fix the issue.
```
Comment (meganrogge) [inline src/vs/platform/terminal/node/terminalEnvironment.ts:244]: can duplication in the PATH cause issues?

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh [hunk 3]
```diff
@@ -10,6 +10,8 @@ fi
 
 VSCODE_SHELL_INTEGRATION=1
 
+echo "before $PATH"
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh:13]:
```suggestion
```

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh [hunk 4]
```diff
@@ -10,6 +10,8 @@ fi
...
+
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh:14]:
```suggestion
```

File: src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh [hunk 5]
```diff
@@ -31,10 +33,18 @@ if [ "$VSCODE_INJECTION" == "1" ]; then
...
+		if [ -n "$VSCODE_PATH_PREFIX" ]; then
+			export PATH=$VSCODE_PATH_PREFIX$PATH
+			builtin unset VSCODE_PATH_PREFIX
+		fi
 	fi
 	builtin unset VSCODE_INJECTION
 fi
 
+echo "after $PATH"
+
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh:47]:
```suggestion
```

File: src/vs/platform/terminal/node/terminalEnvironment.ts (hunk 2 above)
Comment (Tyriar) [reply src/vs/platform/terminal/node/terminalEnvironment.ts:244]: It shouldn't cause problems, just a bit ugly if you check the variable.

**Review by meganrogge — APPROVED** [review]

**Pull request merged**